jrnl/args.py

```diff
@@ -17,6 +17,48 @@
 from .plugins import util
 
 
+def deserialize_config_args(input: list) -> dict:
+
+    """
+
+    Convert a two-element list of configuration key-value pair into a flat dict
+
+    :param input: list of configuration keys in dot-notation and their respective values.
+    :type input: list
+    :return: A single level dict of the configuration keys in dot-notation and their respective desired values
+    :rtype: dict
+    """
+
+    assert len(input) == 2
+    runtime_modifications = {}
+
+    cfg_key = input[0]
+    cfg_value = input[1]
+    cfg_value = cfg_value.strip()
+
+    # Convert numbers and booleans
+    if cfg_value.isdigit():
+        cfg_value = int(cfg_value)
+    elif cfg_value.lower() == "true":
+        cfg_value = True
+    elif cfg_value.lower() == "false":
+        cfg_value = False
+
+    runtime_modifications[cfg_key] = cfg_value
+
+    return runtime_modifications
+
+
+class ConfigurationAction(argparse.Action):
```

Comment:
Is this still needed?

Reply:
It is. This is the action for --config-pverride argument. I intended to have an action for config modification operations, i.e. set, unset, and overrides.

Reply:
I've been going back and forth on this, but I don't this deserialization should happen inside of arg parsing. At this point in the control flow, we don't have the config loaded yet, so we can't validate anything against it.

Can you move the function `deserialize_config_args` into config, and take out the arg parsing action?